bftengine/src/bftengine/PersistentStorageImp.cpp

```diff
@@ -812,6 +816,14 @@ bool PersistentStorageImp::getCompletedMarkInCheckWindow(SeqNum seqNum) {
   return readCompletedMarkFromDisk(seqNum);
 }
 
+std::vector<std::uint8_t> PersistentStorageImp::getUserData() const {
+  auto buf = std::vector<std::uint8_t>(2048);
+  auto actualSize = std::uint32_t{0};
+  metadataStorage_->read(USER_DATA, buf.size(), reinterpret_cast<char *>(buf.data()), actualSize);
+  buf.resize(actualSize);
```

Comment:
Consider replacing vector by queue/deque to avoid resizing

Reply:
I am allocating a buffer, because `read()`'s API requires a `char*` buffer to a contiguous sequence and a length. The resize() should just reduce it for user's convenience. Not sure I got what you meant and I might have misunderstood you, please correct me if I am wrong.